lib/src/core/message/components/message_component.dart

```diff
@@ -77,16 +94,22 @@ class MessageComponentEmoji implements IMessageComponentEmoji {
 abstract class IMessageComponent {
   /// The [ComponentType]
   ComponentType get type;
+
+  /// The custom id of this component, set by the user.
+  late final String customId;
+
+  IMessageComponent._(RawApiMap raw) {
+    customId = raw['custom_id'] as String;
+  }
 }
 
 /// Generic container for components that can be attached to message
-abstract class MessageComponent implements IMessageComponent {
-  /// [ComponentType]
+abstract class MessageComponent extends IMessageComponent {
+  /// Type of component.
```

Comment:
Any implementation should go in the non-interface class (without an `I` prefix). Therefore all implementation classes should implement and not extend their interfaces, and any functionality should go in that class.

Reply:
Oops, my bad, you're right, seems I've messed up my head writing too much interfaces..